lib/pbench/server/api/__init__.py

```diff
@@ -75,6 +79,103 @@ def create_app():
     return app
 
 
+def _build_cors_preflight_response():
+    response = make_response()
+    response.headers.add("Access-Control-Allow-Origin", "*")
+    response.headers.add("Access-Control-Allow-Headers", "*")
+    response.headers.add("Access-Control-Allow-Methods", "*")
+    return response
+
+
+def _corsify_actual_response(response):
+    response.headers.add("Access-Control-Allow-Origin", "*")
+    return response
+
+
+class GraphQL(Resource):
+    def post(self):
+        global app
+        graphQL = app.config_graphql.get("server")
+        json_data = request.get_json()
+
+        if not json_data:
+            app.logger.debug("Invalid query object")
+            abort(400, message="Invalid query object in request")
+
+        try:
+            # query GraphQL
+            response = requests.post(graphQL, json=json_data)
+
+            # construct response object
+            response = make_response(response.text)
+            response = _corsify_actual_response(response)
+        except Exception:
+            app.logger.exception("There was something wrong with the POST request.")
+            abort(500, message="There was something wrong with your graphql request")
```

Comment:
Hmm; is this really an "internal error" or should we do better; for example, perhaps requesting a retry if the GraphQL post showed a connection error or timeout? (I know, decoding errors and figuring out what's recoverable and how can be pretty difficult and sometimes nondeterministic; but adding a fixed 500 error rarely makes anyone happy.)